lib/Dialect/BGV/IR/BGVOps.td

```diff
@@ -11,13 +11,24 @@ include "lib/Dialect/LWE/IR/LWETraits.td"
 include "lib/Dialect/Polynomial/IR/PolynomialAttributes.td"
 include "mlir/Interfaces/SideEffectInterfaces.td"
 include "mlir/IR/BuiltinAttributes.td"
+include "mlir/IR/OpAsmInterface.td"
 
 class BGV_Op<string mnemonic, list<Trait> traits = []> :
-        Op<BGV_Dialect, mnemonic, traits> {
+        Op<BGV_Dialect, mnemonic, traits # [OpAsmOpInterface]> {
```

Comment:
```suggestion
        Op<BGV_Dialect, mnemonic, traits # [DeclareOpInterfaceMethods<OpAsmOpInterface, ["getAsmResultNames"]>]> {
```

Reply:
This will define the methods for you, so you can remove the extra class declaration below. Cf. https://github.com/llvm/llvm-project/blob/1557eeda738d7dbe51d2f52fce28a1fd6f5844ce/mlir/include/mlir/Dialect/Mesh/IR/MeshOps.td#L85 for an upstream example

Same for the other td files.

Reply:
I put the `getAsmResultNames` _definition_ here because otherwise I have to instantiate a it for each type in `Ops.cpp`